src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -413,24 +414,121 @@ private static string EmitIndexOfAnyValues(char[] asciiChars, Dictionary<string,
                 _ => $"s_ascii_{hexBitmap.TrimStart('0')}"
             };
 
-            string helperName = $"IndexOfAnyValues_{fieldName}";
-
-            if (!requiredHelpers.ContainsKey(helperName))
+            if (!requiredHelpers.ContainsKey(fieldName))
             {
                 Array.Sort(asciiChars);
 
                 string setLiteral = Literal(new string(asciiChars));
 
-                requiredHelpers.Add(helperName, new string[]
+                requiredHelpers.Add(fieldName, new string[]
                 {
-                    $"/// <summary>Cached data to efficiently search for a character in the set {EscapeXmlComment(setLiteral)}.</summary>",
+                    $"/// <summary>Supports searching for characters in or not in {EscapeXmlComment(setLiteral)}.</summary>",
                     $"internal static readonly IndexOfAnyValues<char> {fieldName} = IndexOfAnyValues.Create({setLiteral});",
                 });
             }
 
             return $"{HelpersTypeName}.{fieldName}";
         }
 
+        private static string EmitIndexOfAnyCustomHelper(string set, Dictionary<string, string[]> requiredHelpers, bool checkOverflow)
+        {
+            // In order to optimize the search for ASCII characters, we use IndexOfAnyValues to vectorize a search
+            // for those characters plus anything non-ASCII (if we find something non-ASCII, we'll fall back to
+            // a sequential walk).  In order to do that search, we actually build up a set for all of the ASCII
+            // characters _not_ contained in the set, and then do a search for the inverse of that, which will be
+            // all of the target ASCII characters and all of non-ASCII.
+            var asciiChars = new List<char>();
+            for (int i = 0; i <= 0x7f; i++)
+            {
+                if (!RegexCharClass.CharInClass((char)i, set))
+                {
+                    asciiChars.Add((char)i);
+                }
+            }
```

Comment:
If all the values are non-ASCII, this will end up doing `IndexOfAnyExcept("all ASCII values")`.
Do we care about such a case enough to use `IndexOfAnyInRange(min, max)` or skip this optimization entirely instead?

Reply:
> Do we care about such a case enough to use IndexOfAnyInRange(min, max)

IndexOfAnyValue.Create will end up using an implementation that itself uses IndexOfAnyInRange(0, 127), right?

Reply:
Yes, it will.

Using an actual min-max range for the set would be better at avoiding breaking out of the vectorized path too soon in some cases, but I'm not sure how common such sets with no ASCII chars are in practice.
Regardless, this PR isn't making such cases any worse.

Reply:
Changed.